gammapy/scripts/image_pipe.py

```diff
@@ -64,9 +67,12 @@ def make_counts(self, obs_id):
         obs = self.data_store.obs(obs_id=obs_id)
         events = obs.events.select_energy(self.energy_band)
         events = events.select_offset(self.offset_band)
-        counts_map.fill(events=events)
-        counts_map.data = counts_map.data.value
-        self.maps["counts"] = counts_map
+        if (len(events) != 0):
```

Comment:
I also want to change that in this PR if you think it is necessary!!
I agree with you that this is quite complex. Ok to add a _has_counts(obs_id) method but in this case I will have also to check that for the exposure so I was afraid that it could be a computation time problem...
Le 14 avr. 2016 à 15:07, Christoph Deil <notifications@github.com> a écrit :

> In gammapy/scripts/image_pipe.py:
> 
> > @@ -64,9 +67,12 @@ def make_counts(self, obs_id):
> >          obs = self.data_store.obs(obs_id=obs_id)
> >          events = obs.events.select_energy(self.energy_band)
> >          events = events.select_offset(self.offset_band)
> > -        counts_map.fill(events=events)
> > -        counts_map.data = counts_map.data.value
> > -        self.maps["counts"] = counts_map
> > +        if (len(events) != 0):
> I think keeping a list of runs to skip and using it across method calls is unnecessarily complex.
> 
> My suggestion would be to just remove this, i.e. process obs with zero counts.
> They should have zero background and everything should work just fine when stacking.
> 
> Also, speed shouldn't be big issue here ... this is premature optimisation, no?
> If you want to avoid computing the background model for runs with zero counts,
> I'd suggest putting the check for zero counts at the start of that function (or in a method _has_counts(obs_id))
> instead of pre-computing and caching these flags in a self.pass_run member.
> 
> Sometimes such caches are needed for performance, but they are often error-prone, especially if implemented in this way, and in this case I don't think it's needed at all. Just checking for zero counts is much faster than background model computation, so it's OK if the zero-counts check is re-computed a few times (if that is even the case).
> 
> OK or am I missing something?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>